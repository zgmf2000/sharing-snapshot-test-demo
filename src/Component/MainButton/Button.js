//@flow
import { Button } from 'react-native';
import React from 'react';

export default props => {
  const { color = "#ff927b", title } = props;
  return <Button color={color} title={title} />;
};
