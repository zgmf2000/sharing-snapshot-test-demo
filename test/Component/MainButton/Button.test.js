import React from 'react';
import {shallow, configure} from 'enzyme';
import renderer from 'react-test-renderer';
import Button from '../../../src/Component/MainButton/Button'
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Button', () => {
    describe('#render', () => {
        it('should render match snapshot', () => {
            const wrapper = renderer.create(<Button title={'Title Of Button'}/>);
            expect(wrapper.toJSON()).toMatchSnapshot();
        });
    });

    it('should dispatch callback when the button is pressed', () => {
        const mockClickFn = jest.fn();
        const wrapper = shallow(<Button onClick={mockClickFn}/>);
        wrapper.props().onPress();

        expect(mockClickFn).toHaveBeenCalledTimes(1);
    });
});
