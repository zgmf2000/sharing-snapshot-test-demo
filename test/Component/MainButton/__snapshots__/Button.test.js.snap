// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button #render should render match snapshot 1`] = `
<View
  accessibilityRole="button"
  accessibilityStates={Array []}
  accessible={true}
  clickable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "padding": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      Title Of Button
    </Text>
  </View>
</View>
`;
